x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/mapper/SemanticTextInferenceResultFieldMapperTests.java

```diff
@@ -0,0 +1,188 @@
+import java.util.Map;
+import java.util.Set;
+
+public class SemanticTextInferenceResultFieldMapperTests extends MetadataMapperTestCase {
+    private record SemanticTextInferenceResults(String fieldName, List<Pair<Map<String, Float>, String>> inferenceResults) {}
+    private record VisitedChildDocInfo(String path, int sparseVectorDims) {}
+
+    @Override
+    protected String fieldName() {
+        return SemanticTextInferenceResultFieldMapper.NAME;
+    }
+
+    @Override
+    protected boolean isConfigurable() {
+        return false;
+    }
+
+    @Override
+    protected void registerParameters(ParameterChecker checker) throws IOException {
+
+    }
+
+    @Override
+    protected Collection<? extends Plugin> getPlugins() {
+        return List.of(new MachineLearning(Settings.EMPTY));
+    }
+
+    public void testSuccessfulParse() throws IOException {
+        final String fieldName1 = randomAlphaOfLengthBetween(5, 15);
+        final String fieldName2 = randomAlphaOfLengthBetween(5, 15);
+
+        DocumentMapper documentMapper = createDocumentMapper(mapping(b -> {
+            addSemanticTextMapping(b, fieldName1, randomAlphaOfLength(8));
+            addSemanticTextMapping(b, fieldName2, randomAlphaOfLength(8));
+        }));
+        ParsedDocument doc = documentMapper.parse(source(b -> addSemanticTextInferenceResults(b, List.of(
+            new SemanticTextInferenceResults(fieldName1, List.of(
+                new Pair<>(Map.of(
+                    "a", randomFloat(),
+                    "b", randomFloat()
+                ),
+                    "a b"
+                ),
+                new Pair<>(Map.of(
+                    "c", randomFloat()
+                ),
+                    "c"
+                )
+            )),
+            new SemanticTextInferenceResults(fieldName2, List.of(
+                new Pair<>(Map.of(
+                    "d", randomFloat(),
+                    "e", randomFloat(),
+                    "f", randomFloat()
+                ),
+                    "d e f"
+                )
+            ))
+        ))));
+
+        Set<VisitedChildDocInfo> visitedChildDocs = new HashSet<>();
+        Set<VisitedChildDocInfo> expectedVisitedChildDocs = Set.of(
+            new VisitedChildDocInfo(fieldName1, 2),
+            new VisitedChildDocInfo(fieldName1, 1),
+            new VisitedChildDocInfo(fieldName2, 3)
+        );
+
+        List<LuceneDocument> luceneDocs = doc.docs();
+        assertEquals(4, luceneDocs.size());
+        assertValidChildDoc(luceneDocs.get(0), doc.rootDoc(), visitedChildDocs);
+        assertValidChildDoc(luceneDocs.get(1), doc.rootDoc(), visitedChildDocs);
+        assertValidChildDoc(luceneDocs.get(2), doc.rootDoc(), visitedChildDocs);
+        assertEquals(doc.rootDoc(), luceneDocs.get(3));
+        assertNull(luceneDocs.get(3).getParent());
+        assertEquals(expectedVisitedChildDocs, visitedChildDocs);
+
+        // TODO: Need to test how the docs are indexed?
```

Comment:
@carlosdelest I added this section to test how the nested docs are indexed, but it adds significant complexity because the order in which the nested docs are indexed is inconsistent. OK to remove this part of the test?

Reply:
I'd say we need to check that, as that's the main goal of this class. Maybe other folks can chime in with a good strategy for that?

Reply:
I applied a solution similar to one used earlier in the test to validate child docs. Let me know what you think!